extensions/authentication-passport/src/__tests__/acceptance/authentication-with-passport-strategy-adapter.acceptance.ts

```diff
@@ -98,10 +98,18 @@ describe('Basic Authentication', () => {
   function verify(username: string, password: string, cb: Function) {
     users.find(username, password, cb);
   }
+
+  function converter(user: UserProfileInDB): UserProfile {
+    let userProfile = Object.assign({}, user, {[securityId]: user.id});
+    delete userProfile.id;
```

Comment:
What if `securityId` is the string `id`? I think the following solution would be safer:

```ts
const userProfile = Object.assign({}, user);
const id = userProfile.id;
delete userProfile.id;
userProfile.securityId = id;
```

Is it actually necessary to remove the id property? Cannot we preserve it so that clients can use `id` instead of `securityId` if they like to?

Reply:
@bajtos The converter function is not a generic one, it's defined for the test case only(for the mock db attached to the user repository), that's why we know the id field is called `id` and is different than symbol  `securityId`. 

The purpose of having a converter function is allowing developers define their custom conversion according to their User model.